pkg/addonmanager/controllers/registration/controller.go

```diff
@@ -155,23 +155,28 @@ func (c *addonRegistrationController) sync(ctx context.Context, syncCtx factory.
 		_, err = addonPatcher.PatchStatus(ctx, managedClusterAddonCopy, managedClusterAddonCopy.Status, managedClusterAddon.Status)
 		return err
 	}
-	configs := registrationOption.CSRConfigurations(managedCluster)
-
-	managedClusterAddonCopy.Status.Registrations = configs
-
-	managedClusterAddonCopy.Status.Namespace = registrationOption.Namespace
-	if len(managedClusterAddonCopy.Spec.InstallNamespace) > 0 {
-		managedClusterAddonCopy.Status.Namespace = managedClusterAddonCopy.Spec.InstallNamespace
-	}
 
+	var agentInstallNamespace string
 	if registrationOption.AgentInstallNamespace != nil {
-		ns, err := registrationOption.AgentInstallNamespace(managedClusterAddonCopy)
+		agentInstallNamespace, err = registrationOption.AgentInstallNamespace(managedClusterAddonCopy)
 		if err != nil {
+			if _, patchErr := addonPatcher.PatchStatus(ctx, managedClusterAddonCopy, managedClusterAddonCopy.Status, managedClusterAddon.Status); patchErr != nil {
```

Comment:
can we update the status supportedConfigs at line 113 if there are supportedConfigs change, otherwise go ahead? which is easier understand that it is updating the supportedConfigs.